File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if(nums.size()==0) return 0;
        unordered_set<int> karth;
        for(int i=0;i<nums.size();i++)
        {
            karth.insert(nums[i]);
        }
        vector<int> headers;
        int res=1;
        for(int i=0;i<nums.size();i++)
        {
            if(karth.find(nums[i]-1)==karth.end())
            {
                headers.push_back(nums[i]);
            }
        }
        for(int i=0;i<headers.size();i++)
        {
            int num=headers[i];
            int count=1;
            while(karth.find(num+1)!=karth.end())
            {
                count++;
                num++;
                res=max(res,count);
            }
            
        }
        return res;
    }
};

File: Data Structures & Algorithms/subsets/submission-1.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        res=[]
        def func(i,temp):
            if i==len(nums):
                res.append(temp.copy())
                return
            func(i+1,temp)
            temp.append(nums[i])
            func(i+1,temp)
            temp.pop()
        func(0,[])
        return res

        